packages/core/src/browser/shell/tab-bars.ts

```diff
@@ -152,7 +154,96 @@ export class TabBarRenderer extends TabBar.Renderer {
             top = `${paddingTop + iconHeight}px`;
         }
         const style: ElementInlineStyle = { width, height, top };
-        return h.div({ className: 'p-TabBar-tabLabel', style }, data.title.label);
+        if (isInSidePanel || (this.tabBar && this.tabBar.titles.length < 2)) {
```

Comment:
I think documenting this check might be helpful so we can more easily understand what it tries to achieve.